src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.DarcLib.Tests/VirtualMonoRepo/VmrPusherTests.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using FluentAssertions;
+using Microsoft.DotNet.Darc.Models.VirtualMonoRepo;
+using Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+using Microsoft.DotNet.Internal.Testing.Utility;
+using Microsoft.Extensions.Logging.Abstractions;
+using Moq;
+using NUnit.Framework;
+
+namespace Microsoft.DotNet.DarcLib.Tests.VirtualMonoRepo;
+
+[TestFixture]
+public class VmrPusherTests
+{
+    private readonly Mock<ISourceManifest> _sourceManifest = new();
+    private const string GraphQLUri = "https://api.github.com/graphql";
+
+    [Test]
+    public void PushingUnexistingCommitThrowsExceptionTest()
+    {
+        _sourceManifest.Reset();
+        _sourceManifest.SetupGet(s => s.Repositories).Returns(new List<RepositoryRecord>());
```

Comment:
I'd expect we set some fake repositories like the `somerepo` one from below.

Reply:
I was returning a fake one but then I realized it doesn't make any difference what you pass as request body, the `MockHttpClientFactory` only compares the URL, status code and http method. I would like to test also how the request looks like but haven't find a way to do it yet